src/compiler/types.ts

```diff
@@ -794,6 +822,44 @@ module ts {
         typeArguments?: NodeArray<TypeNode>;
     }
 
+    export interface JsxElement extends PrimaryExpression {
+        openingElement: JsxOpeningElement;
+        children?: NodeArray<JsxElement | JsxExpression | JsxText>;
+        closingElement?: JsxClosingElement;
+    }
+
+    export interface JsxOpeningElement extends Expression {
+        tagName: EntityName;
+        attributes: NodeArray<JsxAttribute | JsxSpreadAttribute>;
+        isSelfClosing: boolean;
+    }
+
+    export interface JsxAttribute extends Node {
+        name: Identifier;
+        initializer?: Expression;
+    }
+
+    export interface JsxSpreadAttribute extends Node {
+        expression: Expression;
+    }
+
+    export interface JsxClosingElement extends Node {
+        tagName: EntityName;
+    }
+
+    export interface JsxExpression extends Expression {
+        expression?: Expression;
+    }
+
+    export interface JsxText extends Node {
+        _jsxTextExpressionBrand: any;
+    }
+
+    export interface HeritageClauseElement extends TypeNode {

```

Comment:
I think this was deleted in an intermediary commit from where my branch started and then added back in when I merged. It seems unused; I'll remove it.